pkg/apis/v1alpha5/provisioner.go

```diff
@@ -43,25 +43,25 @@ type ProvisionerSpec struct {
 	// have matching tolerations. Additional taints will be created that match
 	// pod tolerations on a per-node basis.
 	// +optional
-	Taints []v1.Taint `json:"taints,omitempty" hash:"set"`
+	Taints []v1.Taint `json:"taints,omitempty"`
```

Comment:
We do not need `hash:"set"` as all slices will be treated as a set. https://github.com/aws/karpenter-core/blob/34e80492869ae1237a6c57aac8bd009ec41b5eb9/pkg/apis/v1alpha5/provisioner.go#L98